### DISCUSSION TITLE
need help with programming slider

### DISCUSSION THREAD

**Discussion by bijlevel** [RomanKubiak/ctrlr#375] [Q&A] [answered]
My slider needs values in the range from -2400 to 2400. That part is easy. But the values in between are a bit unusual. They start at 2400, 2200, 2000 etc., but then at 1000 they in/decrement in steps of 50, so 950, 900, 850 etc. At 200 they in/decrement in steps of 5, like 195, 190 , 185 etc. Then again at 160 they in/decrement in steps of 2. At last at 100 the in/decrement is in steps of 1, but then at 90 it is again in steps of 2. I hope you get the picture. I can probably solve this by many if-then-else and each time setting the in/decrement value, but maybe one of you has an easier solution?

BTW what is the statement for setting the in/decrement value. Is it something like _**panel:getModulatorByName("tkScale"):getComponent():setIncrement(5,false)**_

All help much appreciated!

**Comment by damiensellier** [comment, marked as answer]
Hi,
You need to use a uiFixedSlider for that and in the slider content box you have to put every single value from -2400 to 2400 with the non linear increments. (one value per line)

```
-2400
-2200
-2000
-1800
-1600
-1400
-1200
-1000
-950
-900
-850
-800
- and so on
```
to catch the "mappedValue" use : `mod:getComponent():getProperty("uiFixedSliderContent")`
_(here  : value 0 = mappedValue -2400, value 1 = mappedValue -2200... )_


Or you can use a regular slider assigned to a lookup table with LUA, with all the following corresponding values in a chain.
I talked about this subject (for a different problematic) a while ago, you can take a look at this thread here https://github.com/RomanKubiak/ctrlr/discussions/255
and here https://github.com/RomanKubiak/ctrlr/discussions/275 for the lookup table with LUA.

And if you need to generate long lists of numbers with x increment,  you can use this generator here : https://pinetools.com/generate-list-numbers (courtesy of @dnaldoog)
for exemple : from 0 to 1000 with 5 increment and "," separator is super easy to generate. hope it helps ;)

PS: I never succeed to make the component value interval work properly.

**Reply by dobo365** [reply]
Yep, uiFixedSlider should do the trick. I wonder if there is any limitation in the length of the content box. Hopefully none ;-)

**Reply by damiensellier** [reply]
No  limit, I put +1000 values once via lua and it worked like a charm.

**Comment by bijlevel** [comment]
Thanks for the answer and tip about generating lists with various in/decrements! This uiFixedSlider seems indeed the solution to my "issue" and now I can use it also for the other odd in/decrement tables of my MIDI device.
